lib/module.js

```diff
@@ -509,7 +509,7 @@ Module.prototype._compile = function(content, filename) {
   var wrapper = Module.wrap(content);
 
   var compiledWrapper = vm.runInThisContext(wrapper, {
-    filename: filename,
+    filename: 'file://' + filename,
```

Comment:
I don't think this is a reasonable change. This changes Node.js stack traces in an unacceptable way. Other scenarios, where the file is actually is stdin or an eval are also broken. Examples:
```
ReferenceError: print is not defined
    at Object.<anonymous> (file:///Users/ofrobots/tmp/6691.js:1:71)
    at Module._compile (module.js:539:32)
    at Object.Module._extensions..js (module.js:548:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Function.Module.runMain (module.js:573:10)
    at startup (node.js:164:18)
    at node.js:449:3
```
and
```
throw new Error("hello")
^
Error: hello
    at [stdin]:1:7
    at Object.exports.runInThisContext (vm.js:54:17)
    at Object.<anonymous> (file://[stdin]-wrapper:6:22)
    at Module._compile (module.js:539:32)
    at node.js:333:29
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```